mobile/bindings.go

```diff
@@ -98,40 +95,37 @@ func Start(extraArgs string, unlockerReady, rpcReady Callback) {
 		}
 	}()
 
-	// Finally we start two go routines that will call the provided
-	// callbacks when the RPC servers are ready to accept calls.
-	go func() {
-		<-unlockerListening
-
-		// We must set the TLS certificates in order to properly
-		// authenticate with the wallet unlocker service.
-		auth, err := lnd.AdminAuthOptions(loadedConfig, true)
-		if err != nil {
-			unlockerReady.OnError(err)
-			return
-		}
-
-		// Add the auth options to the listener's dial options.
-		addWalletUnlockerLisDialOption(auth...)
+	// By default we'll apply the admin auth options, which will include
+	// macaroons.
+	setDefaultDialOption(
+		func() ([]grpc.DialOption, error) {
+			return lnd.AdminAuthOptions(loadedConfig, false)
+		},
+	)
 
-		unlockerReady.OnResponse([]byte{})
-	}()
+	// For the WalletUnlocker and StateService, the macaroons might not be
+	// available yet when called, so we use a more restricted set of
+	// options that don't include them.
+	setWalletUnlockerDialOption(
+		func() ([]grpc.DialOption, error) {
+			return lnd.AdminAuthOptions(loadedConfig, true)
+		},
+	)
+	setStateDialOption(
+		func() ([]grpc.DialOption, error) {
+			return lnd.AdminAuthOptions(loadedConfig, true)
+		},
+	)
 
+	// Finally we start a go routine that will call the provided callback
+	// when the RPC server is ready to accept calls.
 	go func() {
-		<-rpcListening
-
-		// Now that the RPC server is ready, we can get the needed
-		// authentication options, and add them to the global dial
-		// options.
-		auth, err := lnd.AdminAuthOptions(loadedConfig, false)
-		if err != nil {
-			rpcReady.OnError(err)
+		select {
+		case <-rpcListening:
```

Comment:
nit: could move `rpcReady.OnResponse([]byte{})` into this case?

Reply:
I think it's preferable to promote the usage of `SubscribeState` instead, but I guess changing this doesn't hurt.

Reply:
It's not really any difference, I prefer the happy path to continue outside the select.